L1Trigger/L1TGlobal/src/CorrCondition.cc

```diff
@@ -449,19 +452,20 @@ const bool l1t::CorrCondition::evaluateCondition(const int bxEval) const {
 		    if(etaBin0<0) etaBin0 = m_gtScales->getEGScales().etaBins.size() + etaBin0;
 //		    LogDebug("L1TGlobal") << "EG0 phi" << phiIndex0 << " eta " << etaIndex0 << " etaBin0 = " << etaBin0 << " et " << etIndex0 << std::endl;
 
+                    etBin0 = etIndex0;
 		    int ssize = m_gtScales->getEGScales().etBins.size();
-		    if (etIndex0 >= ssize){ 
-		      etIndex0 = ssize-1; 			  
+		    if (etBin0 >= ssize){ 
+		      etBin0 = ssize-1; 			  
 		      edm::LogWarning("L1TGlobal") 
-			<< "EG0 hw et" << etIndex0 << " out of scale range.  Setting to maximum.";
+			<< "EG0 hw et" << etBin0 << " out of scale range.  Setting to maximum.";
 		    } 
 	    
 		    // Determine Floating Pt numbers for floating point caluclation
 		    std::pair<double, double> binEdges = m_gtScales->getEGScales().phiBins.at(phiIndex0);
 		    phi0Phy = 0.5*(binEdges.second + binEdges.first);					    
 		    binEdges = m_gtScales->getEGScales().etaBins.at(etaBin0);
 		    eta0Phy = 0.5*(binEdges.second + binEdges.first);		
-		    binEdges = m_gtScales->getEGScales().etBins.at(etIndex0);
+		    binEdges = m_gtScales->getEGScales().etBins.at(etBin0);
```

Comment:
Hi @rekovic - note that since you are checking already for the validity of the etBin0 index, [] will be faster than .at()